pkg/datasource/formatters/json/options.go

```diff
@@ -44,3 +44,9 @@ func WithPretty(val bool, indent string) Option {
 		formatter.indent = indent
 	}
 }
+
+func WithArray(val bool) Option {
+	return func(formatter *Formatter) {
+		formatter.array = val
+	}
+}
```

Comment:
Is the `val` parameter really needed? I suppose `WithArray()` is clear enough to indicate the array is wanted.

Reply:
It allows doing [this](https://github.com/inspektor-gadget/inspektor-gadget/pull/2709/files#diff-8927e1963c71fd5f4eeb8fd5021767e8b96519013ba4357314d0807fd2532a04R249) instead of adding another if statement. See also [here](https://github.com/inspektor-gadget/inspektor-gadget/pull/2652/files#r1550606356), haha 😄

Reply:
I'm not a big fan of this either, but `WithPretty` behaves already like this, so it did the same for consistency.

Reply:
Alright, let's keep using the same then, but seems weird to me.